### PR TITLE
Font parsing

### PR DESCRIPTION
**Pull request by feliwir**

### CODE REVIEW

File: userspace/libraries/libgraphic/font/FontFace.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <libgraphic/font/Glyph.h>
+#include <libio/Reader.h>
+#include <libutils/unicode/Codepoint.h>
+
+namespace Graphic::Font
+{
+enum FontStyle
```
Comment (sleepy-monax) [inline userspace/libraries/libgraphic/font/FontFace.h:9]: Fonts can be BOLD and ITALIC ;)

File: userspace/libraries/libgraphic/font/Glyph.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+#include <libio/Reader.h>
+#include <libmath/Rect.h>
+
+namespace Graphic::Font
+{
+struct Glyph
+{
+    int advance;
+    int ascend;
```
Comment (sleepy-monax) [inline userspace/libraries/libgraphic/font/Glyph.h:11]: IIRC ascend and descend are global to the font

File: userspace/libraries/libgraphic/font/Parser.cpp [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#include <libgraphic/font/Parser.h>
+#include <libgraphic/font/TrueTypeFontFace.h>
+
+ResultOr<RefPtr<Graphic::Font::FontFace>> Graphic::Font::load(IO::Reader &reader)
+{
+    RefPtr<FontFace> result = TrueTypeFontFace::load(reader).unwrap();
```
Comment (sleepy-monax) [inline userspace/libraries/libgraphic/font/Parser.cpp:6]: Why are you unwrapping it, you are losing the Result

File: userspace/libraries/libgraphic/font/TrueTypeFontFace.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#include <libgraphic/font/TrueTypeFontFace.h>
+#include <libio/FourCC.h>
+#include <libio/Skip.h>
+#include <libio/Streams.h>
+
+// Font tags
+constexpr uint32_t TAG_TrueType_1 = IO::make_tag('1', 0, 0, 0);
+constexpr uint32_t TAG_TrueType_Type1 = IO::make_tag('t', 'y', 'p', '1');
+constexpr uint32_t TAG_OpenType_1 = IO::make_tag(0, 1, 0, 0);
+constexpr uint32_t TAG_OpenType_CFF = IO::make_tag('O', 'T', 'T', 'O');
```
Comment (sleepy-monax) [inline userspace/libraries/libgraphic/font/TrueTypeFontFace.cpp:10]: Why not use arrays for that ?

File: userspace/libraries/libgraphic/font/Parser.cpp (hunk 3 above)
Comment (feliwir) [reply userspace/libraries/libgraphic/font/Parser.cpp:6]: It couldn't convert from `ResultOr<RefPtr<Graphic::Font::TrueTypeFontFace>>` to  `ResultOr<RefPtr<Graphic::Font::FontFace>>`

**Pull request closed without merging**